lstchain/reco/dl0_to_dl1.py

```diff
@@ -182,14 +188,28 @@ def r0_to_dl1(input_filename = get_dataset_path('gamma_test_large.simtel.gz'),
     is_simu = source.metadata['is_simulation']
 
     source.allowed_tels = config["allowed_tels"]
-    source.max_events = config["max_events"]
+    if config["max_events"] is not None:
+        source.max_events = config["max_events"]+1
 
     metadata = global_metadata(source)
     write_metadata(metadata, output_filename)
 
-    cal = load_calibrator_from_config(config)
+    cal_mc = load_calibrator_from_config(config)
+
+    
+    # Camera geometry TBD: read from config file?
+    geom = load_camera_geometry()
```

Comment:
The camera geometry is already part of each event
`geom = event.inst.subarray.tel[tel_id].camera`

Reply:
But is it there already for raw data? When is it introduced?

Reply:
Sure.
It is the reader's responsibility to load such information.

Reply:
The lst event source is doing it.

Reply:
> The lst event source is doing it.

Ok, thx. But how does it work when reading a dl1 file? In the dl1 there is the camera description as a table, how do I read it in a geom object?

By the way, @vuillaut , could that be why lstchain_mc_dl1ab.py had issues? It is using LSTCam-002 hardcoded, perhaps it is not the right one?

Reply:
If the order of the pixels did not change, this should not be a problem. I think the only differences were the coordinate frames between 2 → 3 and the pixel rotation in 3 → 4. Coordinate frames would be important for event reconstructoin, pixel rotation is only important for displaying events at the moment.

Reply:
> If the order of the pixels did not change, this should not be a problem. I think the only differences were the coordinate frames between 2 → 3 and the pixel rotation in 3 → 4. Coordinate frames would be important for event reconstructoin, pixel rotation is only important for displaying events at the moment.

True, but I am now dealing with some (muon) MC in which there seems to be a different ordering of pixels w.r.t. LST-002 to LST-004. So I was wondering if that could be it.

Reply:
> > The lst event source is doing it.
> 
> Ok, thx. But how does it work when reading a dl1 file? In the dl1 there is the camera description as a table, how do I read it in a geom object?
> 
> By the way, @vuillaut , could that be why lstchain_mc_dl1ab.py had issues? It is using LSTCam-002 hardcoded, perhaps it is not the right one?

Ok, I figured out how to do it, should have checked before asking!